packages/firestore/src/core/sync_engine.ts

```diff
@@ -602,6 +609,62 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     }
   }
 
+  /**
+   * Takes a snapshot of current mutation queue, and register a user callback
+   * which will be called when all those mutations in the snapshot are either
```

Comment:
What do you mean by "snapshot" here? Snapshot usually means something different in our code base.